### PR TITLE
Fixed some issues and added persistent last_image

### PR DESCRIPTION
**Pull request by tognee**
creates a last_image file that contains the last image id. Loads the file on startup and updates it when a new image is found

### CODE REVIEW

File: src/instagram/scraper.py [hunk 1]
```diff
@@ -19,8 +22,10 @@ def __get_instagram_feed(username: str) -> requests.Response:
             'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.11 (KHTML, like Gecko) Chrome/23.0.1271.64 '
                           'Safari/537.11 '
         }
-
-        return requests.get(INSTAGRAM_URL + username + '/feed/?__a=1', headers=headers)
+        try:
+            return requests.get(INSTAGRAM_URL + username + '/feed/?__a=1', headers=headers)
+        except requests.exceptions.ConnectionError:
+            return Scraper.__get_instagram_feed(username)
```
Comment (AltF02) [inline src/instagram/scraper.py:30]: Could you possibly add a sleep here so we don't get rate limited?

**Review by AltF02 — CHANGES_REQUESTED** [review]

File: src/instagram/scraper.py (hunk 1 above)
Comment (tognee) [reply src/instagram/scraper.py:30]: Added

**Comment by AltF02** [comment]
Thanks!

**Review by AltF02 — COMMENTED** [review]
LGTM

**Pull request merged**